pkg/conversion/mapi2capi/powervs_fuzz_test.go

```diff
@@ -0,0 +1,186 @@
+
+			return capi2mapi.FromMachineSetAndPowerVSMachineTemplateAndPowerVSCluster(machineSet, powerVSMachineTemplate, powerVSCluster)
+		}
+
+		conversiontest.MAPI2CAPIMachineSetRoundTripFuzzTest(
+			scheme,
+			infra,
+			infraCluster,
+			mapi2capi.FromPowerVSMachineSetAndInfra,
+			fromMachineSetAndPowerVSMachineTemplateAndPowerVSCluster,
+			conversiontest.ObjectMetaFuzzerFuncs(mapiNamespace),
+			conversiontest.MAPIMachineFuzzerFuncs(&mapiv1.PowerVSMachineProviderConfig{}, powerVSProviderIDFuzzer),
+			conversiontest.MAPIMachineSetFuzzerFuncs(),
+			powerVSProviderSpecFuzzerFuncs,
+		)
+	})
+})
+
+func powerVSProviderIDFuzzer(c fuzz.Continue) string {
+	// Power VS provider id format: ibmpowervs://<region>/<zone>/<service_instance_id>/<instance_id>
+	return fmt.Sprintf("ibmpowervs://tok/tok04/%s/%s", strings.ReplaceAll(c.RandString(), "/", ""), strings.ReplaceAll(c.RandString(), "/", ""))
+}
+
+//nolint:funlen
+func powerVSProviderSpecFuzzerFuncs(codecs runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		func(serviceInstance *mapiv1.PowerVSResource, c fuzz.Continue) {
+			switch c.Int31n(3) {
+			case 0:
+				*serviceInstance = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeID,
+					ID:   ptr.To(c.RandString()),
+				}
+			case 1:
+				*serviceInstance = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeID,
+					ID:   ptr.To(c.RandString()),
+				}
+			case 2:
+				*serviceInstance = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeID,
+					ID:   ptr.To(c.RandString()),
+				}
+			}
+		},
+		func(image *mapiv1.PowerVSResource, c fuzz.Continue) {
+			switch c.Int31n(3) {
+			case 0:
+				*image = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeID,
+					ID:   ptr.To(c.RandString()),
+				}
+			case 1:
+				*image = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeName,
+					Name: ptr.To(c.RandString()),
+				}
+			case 2:
+				*image = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeRegEx,
+					Name: ptr.To(c.RandString()),
+				}
+			}
+		},
+		func(network *mapiv1.PowerVSResource, c fuzz.Continue) {
+			switch c.Int31n(3) {
+			case 0:
+				*network = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeID,
+					ID:   ptr.To(c.RandString()),
+				}
+			case 1:
+				*network = mapiv1.PowerVSResource{
+					Type: mapiv1.PowerVSResourceTypeName,
+					Name: ptr.To(c.RandString()),
+				}
+			case 2:
+				*network = mapiv1.PowerVSResource{
+					Type:  mapiv1.PowerVSResourceTypeRegEx,
+					RegEx: ptr.To(c.RandString()),
+				}
+			}
+		},
+		func(pc *mapiv1.PowerVSMachineProviderConfig, c fuzz.Continue) {
+			c.FuzzNoCustom(pc)
+
+			// The type meta is always set to these values by the conversion.
+			pc.Kind = "PowerVSMachineProviderConfig"
+			pc.APIVersion = mapiv1.SchemeGroupVersion.String()
+
+			pc.LoadBalancers = nil
+			pc.ObjectMeta = metav1.ObjectMeta{}
+			pc.CredentialsSecret = nil
```

Comment:
Should there be errors in the conversions if these fields are set? I don't see those